### PR TITLE
Use Airbnb Linting

### PR DESCRIPTION
**Pull request by steets250**
I used this script for setting up eslint and prettier : [https://github.com/paulolramos/eslint-prettier-airbnb-react](https://github.com/paulolramos/eslint-prettier-airbnb-react)

### CODE REVIEW

**Comment by steets250** [comment]
You can ignore commits up to and including "Complete Basic Linting Corrections", as these are purely code changes and would have no affect on the site.

File: package.json [hunk 1]
```diff
@@ -5,17 +5,21 @@
   "proxy": "https://acmucsd-membership-portal-api.herokuapp.com",
   "dependencies": {
     "@ant-design/icons": "^4.0.6",
+    "@firebase/app": "^0.6.3",
```
Comment (leestanley) [inline package.json:8]: Why is there a firebase bump here

Comment (steets250) [reply package.json:8]: It marked it as an implied dependency and required that it was present in the package.json.

File: src/components/EditEventForm/index.jsx [hunk 2]
```diff
@@ -24,26 +23,37 @@ const months = [
...
 
-let hours = [];
-for (let i = 1; i <= 12; i++) {
+const hours = [];
+for (let i = 1; i <= 12; i += 1) {
   hours.push(i);
 }
 
-const EditEventForm = props => {
+const EditEventForm = (props) => {
```
Comment (leestanley) [inline src/components/EditEventForm/index.jsx:36]: ngl we needa put this in a utils.js or something but we'll refactor later

Comment (steets250) [reply src/components/EditEventForm/index.jsx:36]: I agree. I didn't want to rearrange code more than it required and risk breaking anything.

File: src/utils.js [hunk 3]
```diff
@@ -12,18 +12,46 @@ export const notify = (title, description) => {
...
-  const getRandom = list => {
+  const getRandom = (list) => {
     return list[Math.floor(Math.random() * list.length)];
   };
 
-  return `https://api.adorable.io/avatars/face/${getRandom(eyes)}/${getRandom(nose)}/${getRandom(mouth)}/${getRandom(
-    colors
-  )}`;
+  return `https://api.adorable.io/avatars/face/${getRandom(eyes)}/${getRandom(
+    nose
```
Comment (StoneT2000) [inline src/utils.js:53]: airbnb lint makes you do this?? wierd line break

Comment (steets250) [reply src/utils.js:53]: Hmmm, I thought I got rid of the line limit. I'll try switching it back and see what it says.

File: src/index.js [hunk 4]
```diff
@@ -54,27 +53,67 @@ const App = () => {
...
+              exact
+              path="/resendEmailVerification"
+              component={requireAuth(ResendEmailVerificationPage)}
+            />
             <Route
               exact
               path="/secret-bread"
               component={() => {
-                if (easterEggState == 'secret bread') {
+                if (easterEggState === 'secret bread') {
```
Comment (StoneT2000) [inline src/index.js:116]: i'll abstract this easter egg type of access stuff in some later PR

File: src/utils.js (hunk 3 above)
Comment (steets250) [reply src/utils.js:53]: It's showing it as a prettier error. I'll split it at a cleaner spot though.

File: src/containers/requireAuth.jsx [hunk 5]
```diff
@@ -15,21 +14,24 @@ const withAuth = Component => props => {
   }, []);
 
   // TODO: Make redirecting screen and return that if not authenticated.
-  return <Component {...props} />;
+  return <Component />;
```
Comment (StoneT2000) [inline src/containers/requireAuth.jsx:17]: potentially a problem if we don't pass in the props to the component

**Comment by leestanley** [comment]
Let's also switch base to nightly

**Pull request merged**